lib/AST/Module.cpp

```diff
@@ -1391,12 +1395,60 @@ void SourceFile::addImports(
 }
 
 bool SourceFile::hasTestableImport(const swift::ModuleDecl *module) const {
-  using ImportPair = std::pair<ModuleDecl::ImportedModule, ImportOptions>;
+  using ImportPair = std::pair<ModuleDecl::ImportedModule,
+                               std::pair<ImportOptions, StringRef>>;
+  return std::any_of(
+      Imports.begin(), Imports.end(), [module](ImportPair importPair) -> bool {
+        return importPair.first.second == module &&
+               importPair.second.first.contains(ImportFlags::Testable);
+      });
+}
+
+bool SourceFile::hasPrivateImport(AccessLevel accessLevel,
+                                  const swift::ValueDecl *ofDecl) const {
+  using ImportPair = std::pair<ModuleDecl::ImportedModule,
+                               std::pair<ImportOptions, StringRef>>;
+
+  auto *module = ofDecl->getModuleContext();
+
+  if (accessLevel == AccessLevel::Internal ||
+      accessLevel == AccessLevel::Public) {
```

Comment:
And `Open` doesn't need any of this. Please use a `switch` here so that we don't miss cases!